presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TranslateExpressions.java

```diff
@@ -167,15 +172,25 @@ private RowExpression toRowExpression(Expression expression, Session session, Ma
                 return SqlToRowExpressionTranslator.translate(expression, types, ImmutableMap.of(), metadata.getFunctionAndTypeManager().getFunctionAndTypeResolver(), session);
             }
 
-            private RowExpression removeOriginalExpression(RowExpression expression, Rule.Context context)
+            private RowExpression removeOriginalExpression(RowExpression rowExpression, Rule.Context context)
             {
-                if (isExpression(expression)) {
+                if (isExpression(rowExpression)) {
+                    Expression expression = castToExpression(rowExpression);
```

Comment:
Can you remind me which changes caused this additional handling?

Reply:
See the comment in SubqueryPlanner. Its for optimizer rule TransformExistsApplyToLateralNode